truffle/test/Stoppable.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const Stoppable = artifacts.require('Stoppable');
+const truffleAssert = require('truffle-assertions');
+
+contract('Stoppable', (accounts) => {
+    let instance;
+
+    // Initiating new contract before each test
+    beforeEach("Creating new contract", async () => {
+        await Stoppable.new()
+            .then(_instance => {
+                instance = _instance;
+            });
+    });
+
+    // Testing that the function resumeContract is reverted if the contract is already running.
+    it('The contract is already running. resumeContract function should fail', async () => {
+        return await truffleAssert.reverts(
+            instance.resumeContract()
+        );                  
+    });
+
+    // Testing that the function pauseContract is reverted if the contract is already paused.
+    it('The contract is already paused. pauseContract function should fail', () => {
+
+        return instance.pauseContract()
+            .then(async (txObj) => {
+                assert.strictEqual(txObj.logs.length, 1);
+                assert.strictEqual(txObj.logs[0].event, "LogPausedContract");
+
+                await truffleAssert.reverts(
```

Comment:
You like to mix and match `return promise` and `await... You can visually simplify the lot with one or the other.

Reply:
```javascript
const txObj = await instance.pauseContract();
assert.strictEqual(txObj.logs.length, 1);
...
```